lib/cocoapods/installer/project_cache/target_cache_key.rb

```diff
@@ -161,8 +161,8 @@ def self.from_aggregate_target(sandbox, aggregate_target)
           contents = {
             'BUILD_SETTINGS_CHECKSUM' => build_settings,
           }
-          if aggregate_target.includes_resources?
-            relative_file_paths = aggregate_target.resource_paths_by_config.values.flatten.uniq
+          if aggregate_target.includes_resources? || aggregate_target.includes_on_demand_resources?
```

Comment:
this ensures incremental installation works by accounting for on demand resource file references.

If they change then the aggregate target is marked as dirty and is re-integrated, otherwise it is skipped.

Reply:
tests were added below.